flybywiresim/a32nx
# fix(bleed): Improve bleed system stability on long time steps

Fixes #[issue_no]

## Summary of Changes

A few people reported issues of the blue hydraulic pump reporting a fault light on the blue pump. The fault light was explained by the entire bleed system being full of `NaN` values.

While the exact nature of this problem is not entirely clear, we suspect this to be an issue with large timesteps causing issues due to unphysically large pressure and temperature corrections. This PR limits the timestep size by splitting them up into smaller updates.

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): BlueberryKing#6641

## Testing instructions

- Open the Local variable debugger window.
  - Enable dev mode -> Windows -> Behaviors -> Local Variables -> Filter for "PNEU"
- Make sure the bleed and the hydraulic system still works
  - Try starting an engine, enabling and disabling packs, bleed pushbuttons, things like that
  - If you see `nan` for any of the variables during the test, please tell me :)
- Report any noticeable performance degradation.
- Do an engine startup and shutdown


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
